src/util/postgres_client.js

```diff
@@ -478,6 +480,79 @@ class BulkOp {
     }
 }
 
+async function multi_collection_bulk(bulk_per_collection) {
+
+    const client = PostgresClient.instance();
+
+    const queries = _.flatten(Object.values(bulk_per_collection).map(bulk => bulk.get_queries()));
+    const transaction = new PgTransaction(client);
+
+    let ok = false;
+    let errmsg;
+    let nInserted = 0;
+    let nMatched = 0;
+    let nModified = 0;
+    let nRemoved = 0;
+    let should_rollback = false;
+    try {
+        await transaction.begin();
+        should_rollback = true;
+        const batch_query = queries.join('; ');
+        let results = await transaction.query(batch_query);
+        if (!Array.isArray(results)) {
+            results = [results];
+        }
+        for (const res of results) {
+            if (res.command === 'UPDATE') {
+                nModified += res.rowCount;
+                nMatched += res.rowCount;
+            } else if (res.command === 'INSERT') {
+                nInserted += res.rowCount;
+            } else if (res.command === 'DELETE') {
+                nRemoved += res.rowCount;
+            }
+        }
+        await transaction.commit();
+        ok = true;
+    } catch (err) {
+        errmsg = err;
+        dbg.error('PgTransaction execute error', err);
+        if (should_rollback) await transaction.rollback();
+    } finally {
+        transaction.release();
+    }
+    return [errmsg, ok];
+    // return {
```

Comment:
can we clean this comment, or do we want to keep it?

Reply:
in addition to my comment [here](https://github.com/noobaa/noobaa-core/pull/7249#discussion_r1155319425), we should return here the same structure that we return for mongo and handle it in a common code

Reply:
Done.